contracts/PrivateOffer.sol

```diff
@@ -84,11 +84,28 @@ contract PrivateOffer {
         );
 
         // rounding up to the next whole number. Investor is charged up to one currency bit more in case of a fractional currency bit.
-        uint256 currencyAmount = Math.ceilDiv(
+        uint256 maxCurrencyAmount = Math.ceilDiv(
             _arguments.tokenAmount * _arguments.tokenPrice,
             10 ** _arguments.token.decimals()
         );
 
+        // check the allowance of the currencyPayer
+        uint256 allowance = _arguments.currency.allowance(_arguments.currencyPayer, address(this));
+
+        // calculate the token amount and currency amount that can be bought
+        uint256 tokenAmount;
+        uint256 currencyAmount;
+        if(allowance < maxCurrencyAmount) {
+            // the buyer wants to buy less than the offered amount. Calculate the token amount that can be bought with the allowance
+            // mulDiv is used to avoid overflow. It ensures result is rounded down.
+            tokenAmount = Math.mulDiv(allowance, 10 ** _arguments.token.decimals(), _arguments.tokenPrice);
+            currencyAmount = allowance;
+        } else {
```

Comment:
If you remove line 87, and stick with currenyAmount as a name, the else branch isn't needed.